cloudinit/config/schemas/schema-cloud-config-v1.json

```diff
@@ -292,28 +292,69 @@
           "type": "string"
         },
         "lock-passwd": {
-          "default": true,
-          "type": "boolean",
-          "description": "Default: ``true``",
-          "deprecated": true,
-          "deprecated_version": "22.3",
-          "deprecated_description": "Use ``lock_passwd`` instead."
+          "allOf": [
```

Comment:
Why shift the definition under an `allOf`?

Reply:
@holmanb Thanks for questioning this. I originally thought it was to better represent ordering of documentation content on the CLI that ordered the placement of "default" values before the `Deprecated in version 24.2.` documentation. But, I just dumped full schema docs `cloud-init schema --docs all` with allOf schema declaration vs just combining both deprecated* keys and the property constraints and there is no significant difference between either cli-based documentation dumps for with and without the allOf. So, I've avoided this allOf declaration as it equates to the much simpler schema definitions which have the property type constraints and deprecation constraints at the same depth in the schema.
```bash
$ diff cli-docs-with-all-of.out cli-docs-without-all-of.out
546c546
<       **Activate only on keys:** ``ubuntu_pro``, ``ubuntu_advantage``, ``ubuntu-advantage``
---
>       **Activate only on keys:** ``ubuntu_pro``, ``ubuntu-advantage``, ``ubuntu_advantage``
```

Reply:
I've pushed this simpler, flat schema defs for this changeset now to add the coverage needed for the remaining hyphenated key support for no-create-home, plain-text-passwd, hashed-passwd, no-user-group, selinux-user and ssh-redirect-user.